deps/cjs-module-lexer/include/cjs-module-lexer.h

```diff
@@ -0,0 +1,102 @@
+#ifndef __CJS_MODULE_LEXER_H__
+#define __CJS_MODULE_LEXER_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+
+struct Slice {
+  const uint16_t* start;
+  const uint16_t* end;
+};
+typedef struct Slice Slice;
+
+struct StarExportBinding {
+  const uint16_t* specifier_start;
+  const uint16_t* specifier_end;
+  const uint16_t* id_start;
+  const uint16_t* id_end;
+};
+typedef struct StarExportBinding StarExportBinding;
+
+void bail (uint32_t err);
+
+bool parseCJS (uint16_t* source, uint32_t sourceLen, void (*addExport)(const uint16_t*, const uint16_t*), void (*addReexport)(const uint16_t*, const uint16_t*));
+
+void tryBacktrackAddStarExportBinding (uint16_t* pos);
+bool tryParseRequire (bool directStarExport);
```

Comment:
Rather than using `bool` for arguments here, I'd much rather see enum flags... e.g.

```c
enum ParseRequireFlags {
  kParseRequestFlagNone = 0;
  kParseRequestDirectStarExport = 1;
}

bool tryParseRequire(int flags);

// when called

tryParseRequire(kParseRequestDirectStarExport);
```

It makes the call-site a bit easier to read through